tests/jax/test_custom_partitioning.py

```diff
@@ -0,0 +1,301 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+
+import jax
+from jax import random
+import jax.numpy as jnp
+import numpy as np
+import pytest
+from functools import partial
+from jax.experimental.pjit import pjit
+
+from utils import is_devices_enough
+from flax.linen import make_attention_mask
+from flax.linen import make_causal_mask
+from flax.linen import combine_masks
+from transformer_engine.jax.flax import extend_logical_axis_rules
+from transformer_engine.jax.sharding import get_dot_sharding_meta
+from transformer_engine.jax.sharding import get_elementwise_sharding_meta
+from transformer_engine.jax.sharding import get_fp8_meta_sharding_meta
+from transformer_engine.jax.sharding import global_shard_guard
+from transformer_engine.jax.sharding import infer_major_sharding_type
+from transformer_engine.jax.sharding import is_dp_enabled, is_tp_enabled
+from transformer_engine.jax.sharding import ShardingMeta, ShardingResource, ShardingType
+from transformer_engine.jax.sharding import get_elementwise_sharding_meta, extend_fsdp_sharding_meta
+from transformer_engine.jax.sharding_new import MeshResource
+from transformer_engine.jax.sharding_new import global_shard_guard as global_shard_guard_new
+from transformer_engine.jax.layernorm import layernorm
+from transformer_engine.jax.softmax import SoftMaxType, softmax
+from transformer_engine.jax.fused_attn import AttnBiasType, AttnMaskType, self_fused_attn
+from jax.sharding import PartitionSpec, NamedSharding
+P = PartitionSpec
+
+def _get_sharding_spec(mesh_names, sharding_type):
+
+    if sharding_type is ShardingType.DP:
+        return P(mesh_names[0], None), P(None), P(None)
+    elif sharding_type is ShardingType.DP_TP_COL:
+        return P(mesh_names[0], mesh_names[1]), P(None), P(None)
+    else:
+        raise NotImplementedError
+
+def _get_sharding_resource(mesh_names, sharding_type):
```

Comment:
As above description, we are deprecating `ShardingType`, then this might break in the nearly feature.